lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -285,7 +285,11 @@ protected function validateRuntimeMetadata($class, $parent)
                 }
             } else {
                 assert($parent instanceof ClassMetadataInfo); // https://github.com/doctrine/orm/issues/8746
-                if ((! $class->reflClass || ! $class->reflClass->isAbstract()) && ! in_array($class->name, $parent->discriminatorMap)) {
```

Comment:
Do we need to check both? When `validateRuntimeMetadata` is called after the copy of discriminator map from parent to child, then the change becomes easier:

https://github.com/doctrine/orm/blob/2.9.x/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php#L129